internal/pkg/suppression/analyzer.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package suppression defines an analyzer that identifies calls
+// suppressed by a comment.
+package suppression
+
+import (
+	"go/ast"
+	"reflect"
+	"strings"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+// ResultType is a set of nodes that are suppressed by a comment.
+type ResultType map[ast.Node]bool
+
+// IsSuppressed determines whether the given node is suppressed.
+func (rt ResultType) IsSuppressed(n ast.Node) bool {
+	_, ok := rt[n]
+	return ok
+}
+
+var Analyzer = &analysis.Analyzer{
+	Name:       "suppression",
+	Doc:        "This analyzer identifies ast nodes that are suppressed by comments.",
+	Run:        run,
+	ResultType: reflect.TypeOf(new(ResultType)).Elem(),
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	result := map[ast.Node]bool{}
+
+	for _, f := range pass.Files {
+		for node, commentGroups := range ast.NewCommentMap(pass.Fset, f, f.Comments) {
+			for _, cg := range commentGroups {
+				if isSuppressingCommentGroup(cg) {
+					result[node] = true
+					pass.Reportf(node.Pos(), "suppressed")
+				}
+			}
+		}
+	}
+
+	return ResultType(result), nil
```

Comment:
Why cast the result type here instead of initialing the result to be ResultType in line 44?

Reply:
My thinking is that the exact place we do it doesn't matter *too* much, but it has to be done before returning, so why not do it right at the end. This is how it's done in the existing fieldtags analyzer too: 
```go
	// return all facts accumulated down the current path in the dependency graph
	result := map[types.Object]bool{}
	for _, f := range pass.AllObjectFacts() {
		result[f.Object] = true
	}

	return ResultType(result), nil
```

Do you see any pros/cons to doing it when the `result` variable is initialized on line 44?

Reply:
I'd say the distinction is philosophical.  `ResultType` is essentially the exposed API.  It is not meant to be consumed as a `map`.  The underlying type as a `map` is an implementation detail.

If we _were_ to intend the consumption to be done as a `map`, I'd say that we should either not declare `ResultType` at all, or declare it as an alias, i.e. `type ResultType = map [...]...`.  At which point, the declaration one way or the other would be immaterial.

If we are considering it an implementation detail, then I'd say it comes down to preference as to whether you think those details matter in the moment.  In this method, we are using its map-yness, so it makes sense to me to declare it as a `map` and wrap it in the API layer when `return`ing.  On the other hand, that implementation detail is will in scope within this package, and declaring it as a `ResultType` doesn't prohibit you from using it like a `map`.

Personally, I'd prefer the `result := make(ResultType)` declaration and drop the cast on `return`, but I can see it go either way.

Reply:
That all makes sense to me. If there are no strong feelings here, I'll keep it the way it currently is since that's also how it is in `fieldtags/analyzer.go`.